crates/re_viewer/src/store_hub.rs

```diff
@@ -141,6 +159,75 @@ impl StoreHub {
         self.store_dbs.contains_recording(id)
     }
 
+    /// Persist any in-use blueprints to durable storage.
+    // TODO(2579): implement persistence for web
+    #[cfg(not(target_arch = "wasm32"))]
+    pub fn persist_app_blueprints(&mut self) -> anyhow::Result<()> {
+        // Because we save blueprints based on their `ApplicationId`, we only
+        // save the blueprints referenced by `blueprint_by_app_id`, even though
+        // there may be other Blueprints in the Hub.
+        for (app_id, blueprint_id) in &self.blueprint_by_app_id {
+            let blueprint_path = default_blueprint_path(app_id)?;
+            re_log::debug!("Saving blueprint for {app_id} to {:?}", blueprint_path);
+
+            if let Some(blueprint) = self.store_dbs.blueprint(blueprint_id) {
+                // TODO(jleibs): This check isn't perfect. If we GC the blueprint and then insert
+                // more rows we could end up with a collision.
+                if self.blueprint_last_save.get(blueprint_id) != Some(&blueprint.num_rows()) {
```

Comment:
@teh-cmc num_rows() is close to what I wanted, but not quite... I struggled a bit coming up with an easy API call to get a unique/stable "most recently inserted row". At the moment I'm leaning towards adding this as a bit of additional book-keeping in the MetadataRegistry, but curious if you have a better suggestion.

Reply:
Let's chat after standup!